console/console-init/ui/src/modules/iot-device/components/AddDeviceWithJson/AddDeviceWithJson.tsx

```diff
@@ -43,51 +48,110 @@ const styles = StyleSheet.create({
     paddingLeft: 30
   }
 });
+
+export interface IRegistration {
+  enabled?: boolean;
+  via?: string[];
+  ext?: any;
+  viaGroups?: string[];
+  memberOf?: string[];
+  defaults?: any;
+}
 interface IAddDeviceWithJsonProps {
-  deviceDetail?: string;
-  setDeviceDetail: (detail?: string) => void;
+  deviceDetail?: IDeviceResponse;
   onLeave: () => void;
   onSave: (detail: string) => void;
   allowTemplate?: boolean;
 }
+const getDeviceRegistrationString = (deviceDetail: IDeviceResponse) => {
+  let device: any = { registration: {} };
+  if (deviceDetail.registration) {
+    const {
+      enabled,
+      defaults,
+      memberOf,
+      via,
+      ext,
+      viaGroups
+    } = deviceDetail.registration;
+    const deviceRegistration: IRegistration = {};
+    if (enabled) {
+      deviceRegistration.enabled = enabled;
+    }
+    if (defaults) {
+      deviceRegistration.defaults = JSON.parse(defaults);
```

Comment:
please use custom function for parse and stringify i.e. convertStringToJsonAndValidate , convertJsonToStringAndValidate